src/app/portlet/terra-portlet.component.ts

```diff
@@ -1,18 +1,89 @@
 import {
     Component,
-    Input
+    EventEmitter,
+    Input,
+    OnChanges,
+    Output,
+    SimpleChanges
 } from '@angular/core';
+import {
+    animate,
+    state,
+    style,
+    transition,
+    trigger
+} from "@angular/animations";
 
 @Component({
-               selector: 'terra-portlet',
-               styles:   [require('./terra-portlet.component.scss')],
-               template: require('./terra-portlet.component.html')
-           })
-export class TerraPortletComponent
+    selector: 'terra-portlet',
+    styles:   [require('./terra-portlet.component.scss')],
+    template: require('./terra-portlet.component.html'),
+    animations: [
+        trigger('collapsedState', [
+            state('collapsed', style({
+                height: 0,
+                'padding-top': 0,
+                'padding-bottom': 0
+            })),
+            state('expanded', style({
+                height: '*'
+            })),
+            transition('collapsed <=> expanded', [
+                animate(100)
+            ])
+        ])
+    ]
+})
+export class TerraPortletComponent implements OnChanges
 {
+
     @Input() inputPortletHeader:string;
 
+    @Input() isCollapsable: boolean = false;
+
+    @Input() collapsed: boolean = false;
```

Comment:
bitte collapsed zu inputCollapsed ändern und ggf. in html bzw. scss entsprechend anpassen